cirq/circuits/circuit.py

```diff
@@ -1411,8 +1411,8 @@ def batch_insert_into(self,
             Inserted into a moment index that doesn't exist.
         """
         copy = self.copy()
-        for i, op in insert_intos:
-            copy._moments[i] = copy._moments[i].with_operation(op)
+        for i, insertions in insert_intos:
+            copy._moments[i] = copy._moments[i].with_operations(insertions)
         self._device.validate_circuit(copy)
         self._validate_op_tree_qids(copy)
```

Comment:
Add unit tests exercising the new functionality.

Reply:
Done.